src/main/java/kleebot/ui/Parser.java

```diff
@@ -0,0 +1,110 @@
+package kleebot.ui;
+
+import kleebot.command.*;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+public class Parser {
+
+    public static Command parse(String fullCommand) throws KleeExceptions {
+        String[] parts = fullCommand.split("\\s+");
+        String taskDescription;
+        try {
+            switch (parts[0]) {
+            case "bye":
+                return new ExitCommand();
+            case "todo":
+                if (parts.length <= 1) throw new KleeExceptions(Ui.ErrorMessage.MISSING_DETAILS.getMessage());
```

Comment:
Perhaps use the Egyptian style brackets even for single line if statements. 
I notice this happens throughout the codebase.
```suggestion
                if (parts.length <= 1) {
                    throw new KleeExceptions(Ui.ErrorMessage.MISSING_DETAILS.getMessage());
                }